apps/aecontract/test/aecontract_SUITE.erl

```diff
@@ -334,6 +340,45 @@ create_contract_init_error(_Cfg) ->
                  aec_accounts:balance(aect_test_utils:get_account(PubKey, S2))),
     ok.
 
+create_contract_init_error_the_invalid_instruction(_Cfg) ->
+    create_contract_init_error_illegal_instruction_(16#fe, <<"">>).
+
+create_contract_init_error_illegal_instruction_one_hex_digit(_Cfg) ->
+    create_contract_init_error_illegal_instruction_(16#0c, <<"">>).
+
+create_contract_init_error_illegal_instruction_two_hex_digits(_Cfg) ->
+    create_contract_init_error_illegal_instruction_(16#fc, <<"">>).
+
+create_contract_init_error_illegal_instruction_(OP, ErrReason) when is_binary(ErrReason) ->
+    state(aect_test_utils:new_state()),
+    Acc = call(fun new_account/2, [10000000]),
+    Code = compile_contract(minimal_init),
+    HackedCode = hack_bytecode(Code, OP),
+    Gas = 9000,
+    CreateTxOpts = #{gas => Gas},
+    Opts = CreateTxOpts#{return_return_value => true, return_gas_used => true},
+    {{_, {ReturnType, ReturnValue}}, GasUsed} = call(fun create_contract_with_code/5, [Acc, HackedCode, {}, Opts]),
+    ?assertEqual(error, ReturnType),
+    ?assertEqual(ErrReason, ReturnValue),
+    ?assertEqual(Gas, GasUsed),
+    ok.
+
+hack_bytecode(Code, OP) when is_integer(OP), 0 =< OP, OP =< 255 ->
```

Comment:
Some code duplication with `hack_type`. Might want to abstract the common parts.

Reply:
I will address in new version of PR 1810.